source/blog/2019-03-02-update-on-module-unification-and-octane.md

```diff
@@ -0,0 +1,496 @@
+---
+title: 'Update on Module Unification'
+author: Tom Dale
+tags: Recent Posts, 2019, Ember Octane
+responsive: true
+---
+
+<!-- alex disable fight laid burn -->
+
+Ember's conventions for project layout and file naming are central to the
+developer experience. It's crucial that we get both the technical and
+ergonomic details right. I wanted to provide an update about Module
+Unification and our plans for the file structure in Ember Octane.
+
+**In short, we do not plan to ship Module Unification in Octane**. Instead,
+Octane will ship with today's file system layout, with two changes:
+
+* Co-located components and templates
+* Support for nested components in `<AngleBracket />` invocation syntax
+
+### Co-located Components and Templates
+
+Today, component classes and their associated templates are quite spread out.
+If you have a component in `app/components/user-profile.js`, its template
+lives in `app/templates/components/user-profile.hbs`. This can make
+navigating between the two files time-consuming.
+
+In Octane, we will switch to putting a component's JavaScript and template
+file in the same directory.
+
+For example, if you generate a new component called `user-profile` with
+`ember generate component user-profile`, the blueprint will produce these
+files:
+
+* `app/components/user-profile/component.js`
+* `app/components/user-profile/template.hbs`
+* `tests/integration/components/user-profile-test.js`
+
+This co-located `component.js` and `template.hbs` layout actually already
```

Comment:
FYI - This support was added in https://github.com/ember-cli/ember-resolver/pull/61 waaaaaaaaay back in 2014...